### PR TITLE
Add module docstrings and fix formatting

### PR DESCRIPTION
**Pull request by dstrain115**
- Many modules were missing docstrings.  Added simple ones for all
modules.
- Also fixed various formatting issues.
-- Many matrices were not protected by code blocks, so fixed them
-- circuits docstring gets mangled by markdown, so tried to fix it with lists.

### CODE REVIEW

File: cirq/google/__init__.py [hunk 1]
```diff
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+"""Classes for compiling, serializing, and using Google's Quantum Engine API."""
```
Comment (viathor) [inline cirq/google/__init__.py:15]: This doesn't read well due to missing objects: what are we compiling and serializing? Suggestion:

```python
"""Classes for working with Google's Quantum Engine API."""
```

File: cirq/ops/raw_types.py [hunk 2]
```diff
@@ -510,9 +510,11 @@ def validate_args(self, qubits: Sequence['cirq.Qid']):
 
 @value.value_equality
 class TaggedOperation(Operation):
-    """A specific operation instance that has been identified with a set
-    of Tags for special processing.  This can be initialized with
-    Using Operation.with_tags(tag) or by TaggedOperation(op, tag).
+    """A specific operation instance that has identified with a set of Tags.
```
Comment (viathor) [inline cirq/ops/raw_types.py:513]: Somehow "been" has been lost in translation (between "has" and "identified"). Also, this isn't really identifying the operation, but annotating it. Suggestion:

```python
"""Operation annotated with a set of Tags."""
```

File: cirq/ops/raw_types.py [hunk 3]
```diff
@@ -510,9 +510,11 @@ def validate_args(self, qubits: Sequence['cirq.Qid']):
...
+
+    These Tags can be used for special processing.  TaggedOperations
+    can be initialized with using Operation.with_tags(tag)
+    or by iusing TaggedOperation(op, tag).
```
Comment (viathor) [inline cirq/ops/raw_types.py:517]: "iusing" -> "using"

File: cirq/type_workarounds.py [hunk 4]
```diff
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+"""Module to define NotImplementedType."""
```
Comment (viathor) [inline cirq/type_workarounds.py:15]: "to define" -> "defining"

**Review by viathor — APPROVED** [review]
LGTM with a few minor comments to improve style and grammar.

File: cirq/google/__init__.py (hunk 1 above)
Comment (dstrain115) [reply cirq/google/__init__.py:15]: Done.

File: cirq/ops/raw_types.py (hunk 2 above)
Comment (dstrain115) [reply cirq/ops/raw_types.py:513]: Done.

File: cirq/ops/raw_types.py (hunk 3 above)
Comment (dstrain115) [reply cirq/ops/raw_types.py:517]: Done.

File: cirq/pasqal/__init__.py [hunk 5]
```diff
@@ -11,7 +11,7 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
+"""Devices, qubits, and sampler for Pasqal's neutrali atom device."""
```
Comment (HGSilveri) [inline cirq/pasqal/__init__.py:14]: Just a couple typos here, it should be:
```
"""Devices, qubits, and sampler for Pasqal's neutral atom devices."""
```

**Pull request merged**